docker-compose.yml

```diff
@@ -0,0 +1,16 @@
+version: '3.7'
+
+services:
+  talk-archiver:
+    build:
+      context: .
+    volumes:
+      - node_modules:/app/node_modules
+      - ./:/app
```

Comment:
This is confusing me. It mounts the whole of the local directory to `/app`?

Reply:
This technique allows any changes to files on the local files system will propagate to the running container. This is used to avoid rebuilding the container to inject new files while developing. I.e. you can edit files in your IDE outside of the docker container and the changes are present in the docker container. 

If you prefer not to use, we can mount the `.dist/` and `.cache` directories manually to preserve the files between container restarts.

Reply:
Trying this out, it seems like it's slower than running `yarn dev` locally, and crashes with an out-of-memory error while trying to build the `subjects` pages.

Reply:
it'll depend on your docker settings on your laptop. They'll have resource constraints with CPU / RAM at the docker daemon level. I've allowed 2 CPUs and 9GB of ram on my local docker to get it to build via the conatiner on my laptop, i assume larger projects will not build on my laptop no matter how much ram i give docker. 

Using a container on a large cloud instance via docker will not have these limits, rather it'll proxy to the underlying OS resources. Adding docker here was to allow easy reuse of the system, especially to deploy on a virtual instance in the cloud.

Reply:
Cheers. I bumped Docker to 10G and Science Gossip ran, but took ~40 minutes even with cached requests.
```
Done in 2222.95s.
```